turbinia/turbiniactl.py

```diff
@@ -936,42 +931,32 @@ def process_evidence(
                   credential, args.decryption_keys, exception))
           raise TurbiniaException(msg)
         evidence_.credentials.append((credential_type, credential_data))
-    if args.recipe or args.recipe_path:
-      if (args.jobs_denylist or args.jobs_allowlist or
-          args.filter_patterns_file or args.yara_rules_file):
-        msg = (
-            'Specifying a recipe is incompatible with defining '
-            'jobs allow/deny lists, yara rules or a patterns file separately.')
-        raise TurbiniaException(msg)
 
-      if args.recipe_path:
-        recipe_file = args.recipe_path
-      else:
-        recipe_file = os.path.join(config.RECIPE_FILE_DIR, args.recipe)
-      if not os.path.exists(recipe_file) and not recipe_file.endswith('.yaml'):
-        log.warning(
-            'Could not find recipe file at {0:s}, checking for file '
-            'with .yaml extension'.format(recipe_file))
-        recipe_file = recipe_file + '.yaml'
-      if not os.path.exists(recipe_file):
-        msg = 'Recipe file {0:s} could not be found. Exiting.'.format(
-            recipe_file)
-        raise TurbiniaException(msg)
+    # Recipe pre-condition checks.
+    recipe_helpers.validate_recipe_conditions(args)
+    if not hasattr(args, 'skip_recipe_validation'):
```

Comment:
Just curious, is there a known case where this attribute wouldn't exist?

Reply:
skipe_recipe_validatoin is an optional cli argument, so if it's not specified, the argument will not be part of the args namespace. If we make it non-optional and set a default value that will make this hasattr() check obsolete. Thoughts?

Reply:
Hrm, are you sure about that?  At least in a brief test it seems to be part of the namespace as is:
```
In [2]: import argparse

In [3]: parser = argparse.ArgumentParser()

In [4]: parser.add_argument('-X', '--skip_recipe_validation', action='store_true', help='help test', required=False, default=False)
Out[4]: _StoreTrueAction(option_strings=['-X', '--skip_recipe_validation'], dest='skip_recipe_validation', nargs=0, const=True, default=False, type=None, choices=None, help='help test', metavar=None)

In [5]: import sys

In [6]: args = parser.parse_args(sys.argv[1:])

In [7]: args
Out[7]: Namespace(skip_recipe_validation=False)
```

Reply:
Let me run some further tests on this - maybe something within turbiniactl changes that behavior.